processor/src/host/default.rs

```diff
@@ -154,11 +154,8 @@ where
         self.store.get(node_digest)
     }
 
-    fn on_event(
-        &mut self,
-        process: &ProcessState,
-        event_id: u32,
-    ) -> Result<Vec<AdviceMutation>, EventError> {
+    fn on_event(&mut self, process: &ProcessState) -> Result<Vec<AdviceMutation>, EventError> {
+        let event_id = process.get_stack_item(0).as_int() as u32;
```

Comment:
Similar to one of the other comments: I would rather use `try_into().unwrap()` than `as u32`.